sql/src/main/java/io/crate/expression/reference/sys/check/cluster/TablesNeedUpgradeSysCheck.java

```diff
@@ -85,12 +85,14 @@ public BytesRef description() {
         } catch (Throwable t) {
             result.completeExceptionally(t);
         }
-        return result.whenComplete((tableNames, throwable) -> {
+        return result.handle((tableNames, throwable) -> {
             if (throwable == null) {
                 tablesNeedUpgrade = tableNames;
             } else {
                 logger.error("error while checking for tables that need upgrade", throwable);
             }
+            // `select * from sys.checks` should not fail if an error occurred here, so swallow exception
+            return null;
```

Comment:
Is this `null` handled correctly? Meaning what will a select * from sys.checks will return?

Reply:
The result future it only used to wait for the computation to finish (or fail in the failure case), a succesfull result is never used (It's only exposed as `CompletableFuture<?>` so the consumer doesn't even know what is inside)

Reply:
ok, thanks! Didn't remember exactly how that works, just wanted to make sure.